src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LIST_SIZE = 10;
+    private static final int DEFAULT_MULTIPLIER = 1;
+    private int size;
+    private T[] internalArray;
+
+    public ArrayList() {
+        internalArray = (T[]) new Object[DEFAULT_LIST_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= 0 && size < internalArray.length) {
+            internalArray[size] = value;
+            size++;
+        } else if (size == internalArray.length) {
+            grow();
+            internalArray[size] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound array!");
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index >= 0 && index < size) {
+            resizeIdNeeded();
+            System.arraycopy(internalArray, index, internalArray, index + 1, size - index);
+            internalArray[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound array!");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int j = size;
+        for (int i = 0; i < list.size(); i++, j++) {
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (checkIndex(index)) {
+            return internalArray[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound array!");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (checkIndex(index)) {
+            internalArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound array!");
+        }
```

Comment:
```suggestion
        checkIndex(index))
        internalArray[index] = value;
```
fix also other mthods where you use checkIndex method